app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :user, optional: true
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def recent_comments
+    Comment.where(post_id: id).order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_post_counter
+    User.find(author_id).update(posts_counter: Post.where(author_id:).count)
+  end
```

Comment:
- [ ] You have used a method that works but would suggest that you use `method 2` below because of the following reasons: 

**Method 1**
```ruby
def update_post_counter
    User.find(author_id).update(posts_counter: Post.where(author_id:).count)
  end
```

**Method 2**
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

- [ ] Querying and Database Interaction:

  - Method 1: In the first method, `User.find(author_id)` performs a database query to find the user with the given author_id. This incurs a database query overhead.
  - Method 2: The second method directly uses `author.increment!(:posts_counter)` to increment the posts_counter attribute of the author object. This approach avoids an additional database query.

- [ ] Performance and Efficiency:

    - Method 1: The first method involves querying the database for the user and counting the number of associated posts using `Post.where(author_id:).count`. This can be computationally expensive, especially if there are a large number of posts associated with the user.
    - Method 2: The second method simply increments the `posts_counter` attribute on the author object. This operation is efficient and does not depend on the number of associated posts.

- [ ] Readability and Maintainability:

    - Method 1: The first method's implementation may appear less clear and more error-prone due to the missing author_id value in `Post.where(author_id:)`. It might be a typographical error in your code. Additionally, the method involves multiple steps, making it less concise and readable.
    - Method 2: The second method directly increments the posts_counter attribute on the author object. It is more straightforward and readable, encapsulating the logic within a single line.

Reply:
Fixed,